### PR TITLE
SCLJ-60 Library throws an exception when trying to re-connect

### PR DESCRIPTION
**Pull request by metinavci**

### CODE REVIEW

File: client/build.gradle [hunk 1]
```diff
@@ -29,7 +29,7 @@ dependencies {
     compile 'org.glassfish.tyrus.bundles:tyrus-standalone-client:1.13.1'
     compile 'org.slf4j:slf4j-api:1.7.25'
     compile 'org.jetbrains:annotations:16.0.1'
-    compile 'com.squareup.okhttp3:okhttp:3.12.1'
+    compile 'com.squareup.okhttp3:okhttp:3.13.1'
```
Comment (mgouline) [inline client/build.gradle]: What's the point of including OkHttp as a dependency if it's still not used in the library itself? Where did this even come from?

File: client/src/main/java/com/assemblypayments/spi/Spi.java [hunk 2]
```diff
@@ -1563,9 +1563,14 @@ private void onSpiConnectionStatusChanged(@NotNull Connection.State state) {
                     }
 
                     if (conn == null) return; // This means the instance has been disposed. Aborting.
+                    if (reconnectExecutor == null) {
+                        LOG.warn("reconnectExecutor null. Possibly this is still running after dispose?");
+                        return;
+                    }
+                    LOG.info("Will try to reconnect in 3s...");
```
Comment (ellulpatrick) [inline client/src/main/java/com/assemblypayments/spi/Spi.java]: is it really 3s? maybe you can have the log line use the actual variable.

File: client/build.gradle (hunk 1 above)
Comment (metinavci) [reply client/build.gradle]: it is used for Auto Ip Resolution. @RaymondK-AP said me to implement it.

File: client/src/main/java/com/assemblypayments/spi/Spi.java (hunk 2 above)
Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: We decreased delay time in SPI IOS. That's why i fix the log. Fixed the code line.

File: client/src/main/java/com/assemblypayments/spi/Spi.java [hunk 3]
```diff
@@ -1563,9 +1563,14 @@ private void onSpiConnectionStatusChanged(@NotNull Connection.State state) {
                     }
 
                     if (conn == null) return; // This means the instance has been disposed. Aborting.
+                    if (reconnectExecutor == null) {
+                        LOG.warn("reconnectExecutor null. Possibly this is still running after dispose?");
+                        return;
+                    }
+                    LOG.info(String.format("Will try to reconnect in %ds...", RECONNECTION_TIMEOUT / 1000));
```
Comment (mgouline) [inline client/src/main/java/com/assemblypayments/spi/Spi.java]: You don't need another formatter, just do

```java
LOG.info("Will try to reconnect in {}s...", RECONNECTION_TIMEOUT / 1000);
```

File: client/src/main/java/com/assemblypayments/spi/Spi.java [hunk 4]
```diff
@@ -1973,3 +1978,4 @@ public synchronized GeneralSecurityException getCause() {
         void terminalConfigurationResponse(@NotNull Message message);
     }
 }
+
```
Comment (mgouline) [inline client/src/main/java/com/assemblypayments/spi/Spi.java]: Redundant second newline.

File: client/src/main/java/com/assemblypayments/spi/Spi.java (hunk 3 above)
Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: done

File: client/src/main/java/com/assemblypayments/spi/Spi.java (hunk 4 above)
Comment (metinavci) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: done

File: client/src/main/java/com/assemblypayments/spi/Spi.java (hunk 3 above)
Comment (mgouline) [reply client/src/main/java/com/assemblypayments/spi/Spi.java]: Re-read the code that I pasted. That's not what you have.

File: client/src/main/java/com/assemblypayments/spi/Spi.java [hunk 5]
```diff
@@ -1563,9 +1563,14 @@ private void onSpiConnectionStatusChanged(@NotNull Connection.State state) {
                     }
 
                     if (conn == null) return; // This means the instance has been disposed. Aborting.
+                    if (reconnectExecutor == null) {
+                        LOG.warn("reconnectExecutor null. Possibly this is still running after dispose?");
+                        return;
+                    }
+                    LOG.info("Will try to reconnect in %ds...", RECONNECTION_TIMEOUT / 1000);
```
Comment (mgouline) [inline client/src/main/java/com/assemblypayments/spi/Spi.java]: Re-read the code that I pasted. That's not what you have.

**Pull request merged**